### PR TITLE
[How to? pattern articles] Add media session article

### PR DESCRIPTION
**Pull request by beaufortfrancois**
I've tried to keep this one short to fit with other how-to patterns.
@tomayac Please have a look.

I've tested it on Desktop and Mobile.

image

Live previews:

https://deploy-preview-8993--web-dev-staging.netlify.app/patterns/media/
https://deploy-preview-8993--web-dev-staging.netlify.app/patterns/media/media-session/

### CODE REVIEW

File: src/site/content/en/patterns/media/media-session/index.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+title: How to customize media notifications
+date: 2022-11-03
+authors:
+  - beaufortfrancois
+description: >
+  Learn how to let the user control media playback on the lock screen, in a widget, or with hardware keys.
+height: 800
+---
+
+The [Media Session API](https://w3c.github.io/mediasession/) allows websites to let users know what's currently playing in their browser and control it without returning to the page that launched it. The user experience can be customized through metadata in custom media notifications, media events such as playing, pausing, seeking, and track changing. These customizations are available in several contexts including desktop media hubs, media notifications on mobile, and even on wearable devices.
```
Comment (tomayac) [inline src/site/content/en/patterns/media/media-session/index.md:12]:
```suggestion
The [Media Session API](https://developer.mozilla.org/docs/Web/API/Media_Session_API) allows websites to let users know what's currently playing in their browser and control it without returning to the page that launched it. The user experience can be customized through metadata in custom media notifications, media events such as playing, pausing, seeking, and track changing. These customizations are available in several contexts, including desktop media hubs, media notifications on mobile, and even on wearable devices.
```

File: src/site/content/en/patterns/media/media-session/index.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  <figcaption>Media hub on desktop, media notification on mobile, and a wearable device.</figcaption>
+</figure>
+
+When a website is playing audio or video, users automatically get media notifications either in the notification tray on mobile, or the media hub on desktop. The browser does its best to show appropriate information by using the document's title and the largest icon image it can find. The Media Session API allows you to customize the media notification with some richer media metadata such as the title, artist name, album name, and artwork as shown below.
+
+<figure>
+  {% Img src="image/admin/eiavbbCE6TlI8osR1tYT.jpg", alt="Media Session interfaces illustration.", width="800", height="353" %}
+  <figcaption>Anatomy of a media notification on mobile.</figcaption>
+</figure>
+
```
Comment (tomayac) [inline src/site/content/en/patterns/media/media-session/index.md:27]:
```suggestion
```

File: src/site/content/en/patterns/media/media-session/index.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+The example below shows you how to create a custom media notification and respond to basic user's media actions such as play and pause.
```
Comment (tomayac) [inline src/site/content/en/patterns/media/media-session/index.md:29]:
```suggestion
The example below shows you how to create a custom media notification and respond to basic media actions such as play and pause.
```

File: src/site/content/en/patterns/media/media-session/index.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    { src: "https://via.placeholder.com/96",  sizes: "96x96" },
+    { src: "https://via.placeholder.com/128", sizes: "128x128" },
+    { src: "https://via.placeholder.com/256", sizes: "256x256" },
+    { src: "https://via.placeholder.com/512", sizes: "512x512" },
+  ],
+});
+
+navigator.mediaSession.setActionHandler("play", async () => {
+  // Resume playback
+  await video.play();
```
Comment (tomayac) [inline src/site/content/en/patterns/media/media-session/index.md:48]:
```suggestion
  try {
    await video.play();
  } catch (err) {
    alert(err.message);
    }
```

**Review by tomayac — APPROVED** [review]
LGTM with super tiny nits.

Thanks for writing this!

File: src/site/content/en/patterns/media/media-session/assets/script.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    { src: "https://storage.googleapis.com/media-session/caminandes/artwork-96.png",  sizes: "96x96" },
+    { src: "https://storage.googleapis.com/media-session/caminandes/artwork-128.png", sizes: "128x128" },
+    { src: "https://storage.googleapis.com/media-session/caminandes/artwork-256.png", sizes: "256x256" },
+    { src: "https://storage.googleapis.com/media-session/caminandes/artwork-512.png", sizes: "512x512" },
+  ],
+});
+
+navigator.mediaSession.setActionHandler("play", async () => {
+  // Resume playback
+  await video.play();
```
[the remark this answers is not in the document]
Comment (beaufortfrancois) [reply src/site/content/en/patterns/media/media-session/assets/script.js:16]: I'd rather not add `alert` in patterns as it's not a best practise anymore.
Note that the error will throw either way.

File: src/site/content/en/patterns/media/media-session/demo.njk [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    { src: "https://storage.googleapis.com/media-session/caminandes/artwork-96.png",  sizes: "96x96" },
+    { src: "https://storage.googleapis.com/media-session/caminandes/artwork-128.png", sizes: "128x128" },
+    { src: "https://storage.googleapis.com/media-session/caminandes/artwork-256.png", sizes: "256x256" },
+    { src: "https://storage.googleapis.com/media-session/caminandes/artwork-512.png", sizes: "512x512" },
+  ],
+});
+
+navigator.mediaSession.setActionHandler("play", async () => {
+  // Resume playback
+  await video.play();
```
[the remark this answers is not in the document]
Comment (beaufortfrancois) [reply src/site/content/en/patterns/media/media-session/demo.njk:60]: Ditto

File: src/site/content/en/patterns/media/media-session/index.md (hunk 4 above)
Comment (beaufortfrancois) [reply src/site/content/en/patterns/media/media-session/index.md:47]: Same.

**Comment by tomayac** [comment]
`console.error(err.name, err.message);` is our error handling strategy then `:-)`. Better than `alert()` for sure. Neither is production ready of course. 

Thanks again for writing this!

File: src/site/content/en/patterns/media/media-session/index.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  ],
+});
+
+navigator.mediaSession.setActionHandler("play", async () => {
+  // Resume playback
+  try {
+    await video.play();
+  } catch (err) {
+    console.error(err.name, err.message);
+    }
```
Comment (beaufortfrancois) [inline src/site/content/en/patterns/media/media-session/index.md:51]:
```suggestion
  }
  ```

**Pull request merged**